libconway/src/rle.rs

```diff
@@ -164,6 +164,38 @@ fn digits_to_number(digits: &Vec<char>) -> ConwayResult<usize> {
 }
 
 
+/// Dummy implementation of the `CharGrid` trait that is only used to get the size of a pattern
+/// using `Pattern::to_grid`.
+#[derive(Default)]
+struct PatternSize {
+    pub col: usize,
+    pub row: usize,
+}
+
+impl CharGrid for PatternSize {
+    /// Write a char `ch` to (`col`, `row`).
+    fn write_at_position(&mut self, col: usize, row: usize, _ch: char, _visibility: Option<usize>) {
+        if row > self.row {
+            self.row = row;
+        }
+        if col > self.col {
+            self.col = col;
+        }
+    }
+
+    /// Is `ch` a valid character?
+    fn is_valid(_ch: char) -> bool { true }
+
+    /// Width in cells
+    fn width(&self) -> usize { self.col + 1 }
+
+    /// Height in cells
+    fn height(&self) -> usize { self.row + 1 }
+
+    fn get_run(&self, _col: usize, _row: usize, _visibility: Option<usize>) -> (usize, char) { unimplemented!("PatternSize is write-only"); }
```

Comment:
Would make this into an actual block given that it's a long line.

Reply:
Fixing this.